sky/packages/sky/lib/src/animation/performance.dart

```diff
@@ -58,6 +68,84 @@ abstract class PerformanceView {
   bool get isCompleted => status == PerformanceStatus.completed;
 }
 
+class AlwaysCompletePerformance extends PerformanceView {
+  const AlwaysCompletePerformance();
+
+  void updateVariable(Animatable variable) {
+    variable.setProgress(1.0, AnimationDirection.forward);
+  }
+
+  // this performance never changes state
+  void addListener(PerformanceListener listener) { }
+  void removeListener(PerformanceListener listener) { }
+  void addStatusListener(PerformanceStatusListener listener) { }
+  void removeStatusListener(PerformanceStatusListener listener) { }
+  PerformanceStatus get status => PerformanceStatus.completed;
+  AnimationDirection get direction => AnimationDirection.forward;
+  AnimationDirection get curveDirection => AnimationDirection.forward;
+  double get progress => 1.0;
+}
+const AlwaysCompletePerformance alwaysCompletePerformance = const AlwaysCompletePerformance();
+
+class ReversePerformance extends PerformanceView {
+  ReversePerformance(this.masterPerformance) {
+    masterPerformance.addStatusListener(_statusChangeHandler);
+  }
+
+  final PerformanceView masterPerformance;
+
+  void updateVariable(Animatable variable) {
+    variable.setProgress(progress, curveDirection);
+  }
+
+  void addListener(PerformanceListener listener) {
+    masterPerformance.addListener(listener);
+  }
+  void removeListener(PerformanceListener listener) {
+    masterPerformance.removeListener(listener);
+  }
+
+  final List<PerformanceStatusListener> _statusListeners = new List<PerformanceStatusListener>();
+
+  /// Calls listener every time the status of this performance changes
+  void addStatusListener(PerformanceStatusListener listener) {
+    _statusListeners.add(listener);
+  }
+
+  /// Stops calling the listener every time the status of this performance changes
+  void removeStatusListener(PerformanceStatusListener listener) {
+    _statusListeners.remove(listener);
+  }
+
+  void _statusChangeHandler(PerformanceStatus status) {
+    status = _reverseStatus(status);
+    List<PerformanceStatusListener> localListeners = new List<PerformanceStatusListener>.from(_statusListeners);
+    for (PerformanceStatusListener listener in localListeners)
+      listener(status);
+  }
+
+  PerformanceStatus get status => _reverseStatus(masterPerformance.status);
+  AnimationDirection get direction => _reverseDirection(masterPerformance.direction);
+  AnimationDirection get curveDirection => _reverseDirection(masterPerformance.curveDirection);
+  double get progress => 1.0 - masterPerformance.progress;
+
+  PerformanceStatus _reverseStatus(PerformanceStatus status) {
+    switch (status) {
+      case PerformanceStatus.forward: return PerformanceStatus.reverse;
+      case PerformanceStatus.reverse: return PerformanceStatus.forward;
+      default: return status;
```

Comment:
Why don't we need to flip the other two status values?  Otherwise you'll head forward but end up at dismissed.